### PR TITLE
Fix sysincludedirs for codelite.

### PR DESCRIPTION
**Pull request by Jarod42**
**What does this PR do?**

Implement sysincludedirs for codelite.

**How does this PR change Premake's behavior?**

No changes in Premake core. only in codelite generator.

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [ ] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits

### CODE REVIEW

File: modules/codelite/tests/test_codelite_config.lua [hunk 1]
```diff
@@ -57,11 +57,12 @@
 	end
 
 	function suite.OnProjectCfg_Includes()
+		sysincludedirs { "sysdir", "sysdir2/"}
```
Comment (samsinsane) [inline modules/codelite/tests/test_codelite_config.lua:60]: This test isn't for `sysincludedirs`, create a new test.

File: modules/codelite/tests/test_codelite_config.lua [hunk 2]
```diff
@@ -126,12 +127,16 @@
 
 	function suite.OnProjectCfg_ResInclude()
 		files { "x.rc" }
-		resincludedirs { "dir/" }
+		includedirs { "dir/" }
+		sysincludedirs { "sysdir/" }
```
Comment (samsinsane) [inline modules/codelite/tests/test_codelite_config.lua:131]: This test was for `resincludedirs`, create a new test.

**Review by samsinsane — CHANGES_REQUESTED** [review]
When adding new functionality, you add new tests - you don't just hijack existing tests.

File: modules/codelite/tests/test_codelite_config.lua (hunk 1 above)
Comment (Jarod42) [reply modules/codelite/tests/test_codelite_config.lua]: I saw it as all include directives.
split anyway.

File: modules/codelite/tests/test_codelite_config.lua (hunk 2 above)
Comment (Jarod42) [reply modules/codelite/tests/test_codelite_config.lua:164]: I saw it as all relative to resource include directives.
split anyway.

**Review by samsinsane — APPROVED** [review]

**Pull request merged**